### PR TITLE
Add feature to send OTPs in e-mail verifications. 

### PR DESCRIPTION
**Pull request by shanggeeth**
This is implemented as fix to the https://github.com/wso2/product-is/issues/14808
Applying changes from https://github.com/wso2-extensions/identity-governance/pull/617

Following are the changes introducing through this PR. 

1. Checkbox (Send OTP in E-mail) to configure whether to send a confirmation code or a OTP in verification/confirmation e-mail.
2. Three checkboxes (Use uppercase characters in OTP, Use lowercase characters in OTP, User numbers in OTP) to define OTP pattern for e-mail and SMS OTPs.
3. Input field to include the length for the OTPs.

These changes are introduced to the below flows. 

1. Self Registration
2. Lite Registration
3. Ask Password
4. User Claim Update.
5. Notification based Password Recovery

### CODE REVIEW

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/internal/service/impl/otp/OTPGeneratorImpl.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
```
Comment (lashinijay) [inline components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/internal/service/impl/otp/OTPGeneratorImpl.java:2]: Shall we update the license headers?

Comment (shanggeeth) [reply components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/internal/service/impl/otp/OTPGeneratorImpl.java:2]: This commit is from 2022, should we update the year in this file ?

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/internal/service/impl/otp/OTPGeneratorImpl.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    private static final String OTP_GENERATOR_IMPL = "OTPGeneratorImpl";
+
+    /**
+     * Generate the OTP according to given length and pattern. If pattern is not defined default pattern will be used.
+     * Default pattern is to use all three type characters (uppercase characters, lowercase characters and numerics).
+     *
+     * @param useUppercaseLetters Whether uppercase characters should be used for OTP.
+     * @param useLowercaseLetters Whether lowercase characters should be used for OTP.
+     * @param useNumeric          Whether numeric characters should be used for OTP.
+     * @param otpLength         OTP length.
```
Comment (asha15) [inline components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/internal/service/impl/otp/OTPGeneratorImpl.java:43]:
```suggestion
     * @param otpLength           OTP length.
```

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/internal/service/impl/otp/OTPGeneratorImpl.java (hunk 1 above)
Comment (shanggeeth) [reply components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/internal/service/impl/otp/OTPGeneratorImpl.java]: Changed to 2023 in 37a466ec8bc36eb2f5a3a2d160ff5a557ed99a9d

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/internal/service/impl/otp/OTPGeneratorImpl.java (hunk 2 above)
Comment (shanggeeth) [reply components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/internal/service/impl/otp/OTPGeneratorImpl.java]: Addressed with 37a466ec8bc36eb2f5a3a2d160ff5a557ed99a9d

**Review by nilasini — APPROVED** [review]

**Comment by shanggeeth** [comment]
The failure is expected due to these changes and is fixed with
- https://github.com/wso2/product-is/pull/17203

**Pull request merged**